### ISSUE TITLE
All pharo tests pass even though i didn't solve the exercise

### ISSUE THREAD

**Issue by youve** [exercism/pharo-smalltalk#610]
After failing to find my way around the pharo-vm locally, I decided to try to work in the browser. The hello world, exercise was easy enough, but without knowing smalltalk, I didn't know how to start the Roman Numerals exercise, or even what the class should like or what a pass or a fail would look like. I was faced with an empty page, no skeleton of a class written for me like I was expecting.

So I copy pasted the hello world class into the roman numerals exercise. I was hoping that the test would fail and I would learn what my class should be named and I'd get an error message that I could google. Instead, all tests were green:

https://exercism.org/tracks/pharo-smalltalk/exercises/roman-numerals/solutions/youve

**Comment by glennj** [comment]
I'm sorry you're having a tough time. Unfortunately Exercism is really equipped to teach you about the Pharo IDE. Have you read through [the Pharo docs](https://exercism.org/docs/tracks/pharo-smalltalk) and followed the resources?

Also the Pharo discord will have information for beginners: https://discord.gg/kxtXPbfh

**Comment by youve** [comment]
I did read the docs. The part where I got stuck was that I couldn't find the code inside the IDE to edit it -- I did spend at least ten minutes, probably more clicking everything I could think of and exploring the menus, but I know that's not exercism's job to teach me a language that you don't know anything about, it's okay. I gave it a try and I think I don't want to learn something right now that I can't edit in vim.

The reason I reported it as a bug is that I can submit a solution that doesn't solve the exercise and still get credit for it: the test runner says that all 19 tests passed, but if you look at the code I entered, you can see that it should not pass the tests in the roman numerals problem:

```smalltalk
Class {
        #name : #HelloWorld,
        #superclass : #Object,
        #category : #'Exercise@HelloWorld'
}
{ #category : #example }
HelloWorld class >> sayHello [
        ^ 'Goodbye, Mars!'
]
```

**Comment by Bajger** [comment]
Hi @youve! Thanks for reporting this, I will definitely look at this! Did you submit your exercise via IDE or through Exercism website (code editor)? 

Regarding finding a code: On page there is a paragraph dealing with System Browser - place where you can see how Pharo code is organized. See: https://exercism.org/docs/tracks/pharo-smalltalk/installation
You can type also `shift + ender` this would enable quick search and try to find `hello`, it should give you options to browse package or class. If you find documentation confusing, not clear enough, let me know how to improve this.
**Quick search:**
![obrazek](https://github.com/exercism/pharo-smalltalk/assets/45875448/6c9e284e-96a5-4915-9d82-b1f25c9ece92)

**System browser:**
![obrazek](https://github.com/exercism/pharo-smalltalk/assets/45875448/e14ba546-6d3f-4539-ad28-d7a364b7c502)

**Comment by youve** [comment]
Hi! I submitted it through Exercism website (code editor). The same code snippet works for the other exercises, too - I first noticed it with two-fer but I didn't publish that as a solution.

Thanks for the screenshots, I'll give it another try later today.

**Comment by Bajger** [comment]
It is related to old issue, which still preserves in Pharo. https://github.com/pharo-project/pharo/issues/11155
Problem is that Pharo test runner uses "vanilla" image, that includes all example sources and tests. Source classes are deleted before import of your solution. Problem is in the fact that even after deletion of original class with solution, test keeps remembering deleted class (still refers to that - it is called ObsoleteClass), until new class is recompiled by your source code import. But if your source code doesn't include definition of roman numerals class (you left there HelloWorld class), it keeps using the ObsoleteClass (with all methods from vanilla solution). 

TLDR: I will try to find a way: 
- either by not using image with all example solutions
- or compile dummy RomanNumeral class without any method definition
- or find a way to convince instance of ObsoleteClass to not execute original compiled method

**Comment by Bajger** [comment]
Fix must be applied to test runner, after that will be applicable on Exercism website.

**Comment by Bajger** [comment]
Used wrong package for P11 dependency, most recent PR should fix it.

**Comment by Bajger** [comment]
@ErikSchierboom Hi Erik! Do you know, if workflow actions have been changed recently? I struggle to trigger update of test runner on AWS. On CI.yml I see correct commit, with which was test runner built (correct commit SHA is part of install command) - see: 
https://github.com/exercism/pharo-smalltalk-test-runner/actions/runs/8536327023/job/23384609404#step:5:243 

But I'm not sure, if something have changed, for deployment to AWS. It seems that previous version of test runner (build) is used.
I've tested with latest changes locally and works as expected: 
![image](https://github.com/exercism/pharo-smalltalk/assets/45875448/437c3683-a6bd-4b0e-acf6-0863bab93703)

**Comment by ErikSchierboom** [comment]
No, the deploy seems to have been a regular, succesful deploy: https://github.com/exercism/pharo-smalltalk-test-runner/actions/runs/8536327102/job/23384609988

**Comment by Bajger** [comment]
But it seems to be cached, maybe need to add env. variable with commit hash value? But I don't really know where: https://github.com/exercism/pharo-smalltalk-test-runner/actions/runs/8536327102/job/23384609988#step:7:161

**Comment by ErikSchierboom** [comment]
I've fixed it in https://github.com/exercism/pharo-smalltalk-test-runner/pull/31. As you can see from a test deploy run, the sync commit is now being used when deploying: https://github.com/exercism/pharo-smalltalk-test-runner/actions/runs/8550913841/job/23428825179

**Comment by Bajger** [comment]
@ErikSchierboom Thank you!
@youve issue should be fixed now, Roman Numerals will fail, if you pass code with different exercise:
![image](https://github.com/exercism/pharo-smalltalk/assets/45875448/cf5e74bc-fd15-4365-9e05-a310f82dd969)
